shared/Microsoft.Extensions.ObjectMethodExecutor.Sources/ObjectMethodExecutor.cs

```diff
@@ -354,6 +366,51 @@ private static MethodExecutor WrapVoidMethod(VoidMethodExecutor executor)
             return lambda.Compile();
         }
 
+        private static bool IsAwaitableDirectlyOrViaCoercion(
+            Type type,
+            out Expression coerceToAwaitableExpression,
```

Comment:
Yeah, and that structure could also wrap up a lot of the Expression generation code as well.